modem: hl7800: Fix crash related to IPv6 address

rerickson1 opened the pull request:
If modem receives an IPv6 address, buffer overrun
would occur. Fix this by checking string length to
ensure what type of IP address needs to be parsed.

Correct destination buffer size when processing
RX messages from modem.

jukkar commented on drivers/modem/hl7800.c:
```diff
@@ -255,7 +255,7 @@ static const struct mdm_control_pinconfig pinconfig[] = {
 #define MDM_MAX_DATA_LENGTH 1500
 #define MDM_MTU 1500
-#define MDM_MAX_RESP_SIZE 128
+#define MDM_MAX_RESP_SIZE 256
```
This creates quite high pressure to stack usage, why do we need such a large value here?

jukkar commented on drivers/modem/hl7800.c:
```diff
@@ -1660,7 +1676,7 @@ static bool on_cmd_atcmdinfo_operator_status(struct net_buf **buf, uint16_t len)
 	int i;
 
-	out_len = net_buf_linearize(value, len, *buf, 0, len);
+	out_len = net_buf_linearize(value, sizeof(value), *buf, 0, len);
```
Doesn't this belong to the other commit?

rerickson1 committed:
drivers: modem: hl7800: fix potential buffer overrrun

Correct destination buffer size when processing
RX messages from modem.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>

jukkar commented on drivers/modem/hl7800.c:
```diff
@@ -256,6 +256,7 @@ static const struct mdm_control_pinconfig pinconfig[] = {
 #define MDM_MAX_DATA_LENGTH 1500
 #define MDM_MTU 1500
 #define MDM_MAX_RESP_SIZE 128
+#define MDM_IP_INFO_RESP_SIZE 256
```
I am still wondering why you need to use this long string. How many IPv6 addresses you are expecting to store here? The max length of the IPv6 address string is 46 chars (INET6_ADDRSTRLEN) so the length of 256 looks rather long.

rerickson1 replied:
+CGCONTRDP: response is very long for IPv6. Sierra uses IPv6 address format: `a1.a2.a3.a4.a5.a6.a7.a8.a9.a10.a11.a12.a13.a14.a15.a16`

rerickson1 replied:
(3 * 16) + 15 for each address in the response. Response has at least IP address, IP gateway, and one DNS address.
Thats at least 189 characters in the response not including the other fields.

rlubos reviewed (approved):
The diff looks a bit scary due to the extra indentation, however the aren't many functional changes and they look reasonable to me (I'd consider returning early `if (!is_ipv4)` though, but I guess it's fine like this as well).

The stack consumption seems a bit concerning to me, there's some room for improvement, but I guess it's a story for a different PR.

rerickson1 committed:
drivers: modem: hl7800: fix fault with IPv6 address

If modem receives an IPv6 address, buffer overrun
would occur. Fix this by checking string length to
ensure what type of IP address needs to be parsed.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdconnect.com>
